### DISCUSSION TITLE
HashMap

### DISCUSSION THREAD

**Discussion by nearest-river** [denoland/deno#19965] [Q&A]
does anyone know how the native Map works..
its hashing..

**Comment by boywithkeyboard** [comment]
@kakashi-69-xd if you can't find the answer to your question on either MDN ([this document](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)) or in the [official specification](https://www.ecma-international.org), you're probably out of luck.

**Comment by ghost** [comment]
This isn't relevant to Deno.

But the native JavaScript `globalThis.Map` is implemented on an engine-by-engine basis and is differently implemented in each.

Check the V8 source code or documentation if you want to know how V8 does it. V8 is the JavaScript and WebAssembly engine used by the Deno runtime.

Really you seem confused though. What do you want to actually understand?

**Reply by nearest-river** [reply]
trying to make one faster than the std
